lale/lib/lale/hyperopt_regressor.py

```diff
@@ -86,14 +88,17 @@ def f(params):
                 # if max optimization time set, and we have crossed it, exit optimization completely
                 sys.exit(0)
 
+            return_dict = {}
             try:
-                r_squared, execution_time = hyperopt_train_test(params, X_train=X_train, y_train=y_train)
+                score, execution_time = hyperopt_train_test(params, X_train=X_train, y_train=y_train)
+                return_dict = {
+                    'loss': self.best_score - score, 'time': execution_time, 'status': STATUS_OK
+                }
             except BaseException as e:
-                logger.warning(f'Exception caught in HyperoptRegressor: {type(e)}, {traceback.format_exc()} with hyperparams: {params}, setting loss to zero')
-                r_squared = 0
-                execution_time = 0
-            return {'loss': -r_squared, 'time': execution_time, 'status': STATUS_OK}
 
+                logger.warning(f'Exception caught in HyperoptRegressor: {type(e)}, {traceback.format_exc()} with hyperparams: {params}, setting loss to zero')
+                return_dict = {'status': STATUS_FAIL}
```

Comment:
I have changed this to return a `STATUS_FAIL` since I could not think of a good default loss function for `r2_score` (it can be arbitrarily high).